### PR TITLE
4 tcp server multiple clients

### PR DESCRIPTION
**Pull request by TristanAlbers**
Rudimentary publishing and subcribing with a broker (tcp server). 
Is able to:
- Handle multiple publishers
- Handle multiple subscribers
- Send a message from a publisher to all subscribers

Is not able to:
- Communicate via official mqtt headers
- Differentiate between different topics (no notion of topics has been implemented yet, the server handles as a single topic)
- Differentiate between QoS (MVars automatically provide a QoS of possibly 2, worth looking further in to)
- Error handle on connection drops

This could be a great starting point from which we can build further

References:
[MVars](https://hackage.haskell.org/package/base-4.18.0.0/docs/Control-Concurrent-MVar.html)
[Concurrency](https://www.youtube.com/watch?v=cuHD2qTXxL4)

### CODE REVIEW

**Review by JesseFloren — COMMENTED** [review]
Looks great, important for all devs to understand the usage of MVar.

**Review by JesseFloren — APPROVED** [review]
Approved!!

File: lib/Client.hs [hunk 1]
```diff
@@ -2,21 +2,45 @@ module Client where
...
+    s <- Prelude.getLine
+    sendAll sock $ C.pack s
     -- receive response
     response <- recv sock 1024
     putStrLn $ "Received: " ++ show response
-    
-    -- close connection
-    close sock
+    -- create type of client based on input
+    createClient sock s
```
Comment (JanMouwes) [inline lib/Client.hs:18]: 'createClient' feels like it should yield something, implied by create

**Review by JanMouwes — APPROVED** [review]
LGMT :)

**Comment by JanMouwes** [comment]
> Looks great, important for all devs to understand the usage of MVar.

MVar isn't used yet though, right?

**Pull request merged**